modules/utxo-ord/src/inscriptions.ts

```diff
@@ -57,8 +63,98 @@ export function createOutputScriptForInscription(
     output: compiledScript,
     depth: 0,
   };
-  const payment = payments.p2tr({ redeems: [redeem] }, { eccLib });
+
+  return payments.p2tr({ redeems: [redeem], redeemIndex: 0 }, { eccLib });
+}
+
+/**
+ * Creates a dummy signed transaction to get the size of the reveal transaction
+ * used to calculate fees.
+ * @param network
+ * @param controlBlock
+ * @param commitOutput
+ */
+function getInscriptionRevealSize(
+  payment: Payment,
+  controlBlock: Buffer,
+  commitOutput: Buffer,
+  network: Network
+): number {
+  const psbt = bitgo.createPsbtForNetwork({ network });
+  const parsedControlBlock = taproot.parseControlBlock(eccLib, controlBlock);
+  const leafHash = taproot.getTapleafHash(eccLib, parsedControlBlock, payment.redeem?.output as Buffer);
+
+  psbt.addInput({
+    hash: Buffer.alloc(32),
+    index: 0,
+    witnessUtxo: { script: commitOutput, value: BigInt(10239812038) },
+    tapLeafScript: [
+      {
+        controlBlock,
+        script: payment.redeem?.output as Buffer,
+        leafVersion: taproot.INITIAL_TAPSCRIPT_VERSION,
+      },
+    ],
+  });
+  psbt.addOutput({ script: commitOutput, value: BigInt(10000) });
+
+  psbt.signTaprootInput(
+    0,
+    {
+      publicKey: Buffer.alloc(32),
+      signSchnorr(hash: Buffer): Buffer {
+        return Buffer.alloc(64);
+      },
+    },
+    [leafHash]
+  );
+
+  psbt.finalizeInscriptionInput(0);
+  return psbt.extractTransaction(true).virtualSize();
+}
+
+export type InscriptionRevealData = {
+  address: string;
+  revealSize: number;
+};
+
+/**
+ * @returns InscriptionRevealData
+ * @param pubkey
+ * @param contentType
+ * @param inscriptionData
+ * @param network
+ */
+export function createInscriptionRevealData(
+  pubkey: Buffer,
+  contentType: string,
+  inscriptionData: Buffer,
+  network: Network
+): InscriptionRevealData {
+  const payment = createPaymentForInscription(pubkey, contentType, inscriptionData);
+
+  const { output: commitOutput, controlBlock } = payment;
+  assert(commitOutput);
+  assert(controlBlock);
+  const commitAddress = address.fromOutputScript(commitOutput, network);
+  const revealSize = getInscriptionRevealSize(payment, controlBlock, commitOutput, network);
+
+  return { address: commitAddress, revealSize };
+}
+
+/**
+ * @param pubkey
+ * @param contentType
+ * @param inscriptionData
+ * @returns inscription address
+ */
+export function createOutputScriptForInscription(pubkey: Buffer, contentType: string, inscriptionData: Buffer): Buffer {
+  const payment = createPaymentForInscription(pubkey, contentType, inscriptionData);
 
   assert(payment.output, 'Failed to create inscription output script');
   return payment.output;
 }
+
+export function getFeesForReveal(revealSize: number, feeRate: number): bigint {
+  return BigInt(revealSize * revealSize);
+}
```

Comment:
wrong implementation, missing tests and missing jsdoc (feeRate needs a unit)

given that we don't seem to use this method let's get rid of it. We already have methods to multiple virtual size with feeRate.

```suggestion
```